### ISSUE TITLE
[Enhancement] Create an ETL to sync all Moped project edits with Data Tracker

### ISSUE THREAD

**Issue by johnclary** [cityofaustin/atd-data-tech#8536]
Currently, the "Knack sync" is a one-time push to the AMD Data Tracker knack app. This task is to replace the Knack sync functionality with an ETL that syncs all Moped projects with the AMD Data Tracker.

The ETL only needs to handle the same columns which are currently pushed to the Data Tracker. No additional columns are needed.

### Resources
- the [`projects` object](https://builder.knack.com/atd/amd/schema/list/objects/object_201/fields) in the AMD Data Tracker which holds Moped data
- this [partially complete](https://github.com/cityofaustin/atd-moped/blob/main/moped-etl/data-tracker-sync/app/moped_data-tracker_sync_projects.py) (?) ETL of a previous iteration of this feature
- the [existing Moped editor code](https://github.com/cityofaustin/atd-moped/blob/main/moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryKnackDataTrackerSync.js) which implements the current feature

**Comment by frankhereford** [comment]
While we’re waiting for the heroku maintenance on our testing DB, I have been looking into `graphql-engine` options we have for automatically pushing updated data about moped projects to the Knack Data Tracker application. I’m sure this is all old hat for all of you, but actions are a new feature to me, so I found it helpful to compile some observations when considering routes on how to automate keeping knack in sync with moped.

We have at least two options: Actions and Events. They are related to each other in that they will ultimately call a “serverless”, lambda function to execute an update to knack. We can’t directly use knack as the provider of the endpoint without the lambda function in the middle, because the data & format offered up by `graphql-engine` does not exactly match the data & format of the knack endpoint. We are already running a couple of these lambda functions, so rolling this functionality into the `moped-api` makes sense to me.

Actions are used to extend the graphql syntax exposed to our endpoint, such that the `graphql-engine` server can return information to the moped application from knack in the results of a mutation, à la returning `affected_rows`. Consider that `graphql-engine` is not going to be able to resolve the request from the moped app until it’s done with it’s request to knack. Events have an appeal if we want to be able to write logic in our moped app that may change up the form of our mutation to perhaps only include some columns that are being updated or update the user that a sync has completed.

Events are simpler and easier to configure, in my opinion. These are fired off when a certain database action happens (`update`, `insert`, etc), down to the column granularity. The details of the change and other supporting data are sent off to the lambda function and the result is captured by graphql-engine, but not stored in the `public` schema, just the `hdb_*` ones. These events happen, in our case, always after a mutation, but the app that caused the mutation doesn’t really know or care about this knock-on process stemming from the data update. The lambda function would then be responsible for communicating with knack and submitting the update to data tracker. This pattern is already in use in the moped stack.

My opinion is to lean towards another Event configuration for this purpose. It’s easier to set-up and we don’t really need feedback about the synchronization outcome in the moped application, as it’s intended to be “behind the scenes” to the user. From their point of view, they are updating data in moped, their source of record, and are not probably thinking about data tracker at all. We also get per-column resolution on what triggers the update event, so that’s nice since we’re only pushing a small subset of data over to knack.

Looking forward to your thoughts - both are entirely viable routes, to be sure.

**Comment by amenity** [comment]
Today, from Ivonne via Teams 

> How often the MOPED project list is updated on data tracker? BC was looking for a project on the pull-down list in data tracker and it was not there, project was created a while back.  Someone else reported the same issue.  
> I want data tracker users to identify the MOPED project in their WRK but I am not sure the list is current
> Let me know if this question needs a ticket. 
> 
> For example WRK22-165025 Brian wanted to link to APS project #129 -- we created project 6/29 but I changed project name on 7/28, the name change is not reflected in the pull-down list in data tracker, so BC thought the project was not listed...

**Comment by johnclary** [comment]
Having just tested out Hasura's [REST connectors](https://hasura.io/docs/latest/event-triggers/rest-connectors/) in [this issue](https://github.com/cityofaustin/atd-data-tech/issues/10600), I'm quite sure we can create an event payload that will push updates to the Data Tracker, without the need for the intermediary `moped-api`. 

The initial "Sync with data tracker" step would remain in place. Then, once sync'd, our event listener push changes to the project name (or whatever other columns) automatically. 

TLDR, yes to what Frank recommended, but with REST Connectors plugged in :) I'm bringing the estimate on this down to `3` because this is looking straightforward now 🚀
